src/coreclr/jit/morph.cpp

```diff
@@ -8955,32 +8254,31 @@ GenTree* Compiler::fgGetStubAddrArg(GenTreeCall* call)
 // corresponds to the argument to a recursive call.
 //
 // Notes:
-//    Due to non-standard args this is not just fgArgTabEntry::argNum.
-//    For example, in R2R compilations we will have added a non-standard
-//    arg for the R2R indirection cell.
+//    Due to non-standard args this is not just the index of the argument in
+//    the arg list. For example, in R2R compilations we will have added a
+//    non-standard arg for the R2R indirection cell.
 //
 // Arguments:
-//    argTabEntry  - the arg
+//    arg  - the arg
 //
-unsigned Compiler::fgGetArgTabEntryParameterLclNum(GenTreeCall* call, fgArgTabEntry* argTabEntry)
+unsigned Compiler::fgGetArgParameterLclNum(GenTreeCall* call, CallArg* arg)
 {
-    fgArgInfo*      argInfo  = call->fgArgInfo;
-    unsigned        argCount = argInfo->ArgCount();
-    fgArgTabEntry** argTable = argInfo->ArgTable();
+    unsigned num = arg->AbiInfo.ArgNum;
 
-    unsigned numToRemove = 0;
-    for (unsigned i = 0; i < argCount; i++)
+    for (CallArg& otherArg : call->gtArgs.Args())
     {
-        fgArgTabEntry* arg = argTable[i];
+        if (&otherArg == arg)
+        {
+            break;
+        }
         // Late added args add extra args that do not map to IL parameters and that we should not reassign.
-        if (!arg->isNonStandard() || !arg->isNonStandardArgAddedLate())
+        if (!otherArg.IsArgAddedLate())
```

Comment:
Would this be simpler?
```C++
   if (otherAr.IsArgAddedLate())
   {
        num--;
   }

Reply:
Yes, but let me leave it to my follow up change where I plan to remove `ArgNum` -- this function will need to change then as well.